library/spdm_requester_lib/libspdm_req_get_certificate.c

```diff
@@ -91,11 +94,30 @@ libspdm_return_t libspdm_try_get_certificate(void *context, uint8_t slot_id,
             SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CERT_CAP)) {
         return LIBSPDM_STATUS_UNSUPPORTED_CAP;
     }
-    libspdm_reset_message_buffer_via_request_code(spdm_context, NULL, SPDM_GET_CERTIFICATE);
-    if (spdm_context->connection_info.connection_state <
-        LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
-        return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+    if (session_id == NULL) {
+        if (spdm_context->connection_info.connection_state <
+            LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+            return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+        }
+        session_info = NULL;
+    } else {
+        if (spdm_context->connection_info.connection_state <
+            LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+            return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+        }
```

Comment:
This check is same as the one in if-branch. No need to duplicate. Please move them out of if-else.